src/main/java/seedu/address/model/attendance/AttendanceRecordList.java

```diff
@@ -0,0 +1,141 @@
+    public Attendance getAttendance(Student student, Week week)
+            throws InvalidWeekException, AttendanceNotFoundException {
+        requireAllNonNull(student, week);
+
+        if (!isWeekContained(week)) {
+            throw new InvalidWeekException();
+        }
+
+        UUID studentUuid = student.getUuid();
+        int weekIndex = week.getZeroBasedWeekNumber();
+        return recordList.get(weekIndex).getAttendance(studentUuid);
+    }
+
+    /**
+     * Adds a {@code Student}'s {@code Attendance} to the specified {@code Week}.
+     */
+    public AttendanceRecordList addAttendance(Student student, Week week, Attendance attendance)
+            throws InvalidWeekException, DuplicateAttendanceException {
+        requireAllNonNull(student, week, attendance);
+
+        if (!isWeekContained(week)) {
+            throw new InvalidWeekException();
+        }
+
+        UUID studentUuid = student.getUuid();
+        int weekIndex = week.getZeroBasedWeekNumber();
+        List<AttendanceRecord> updatedRecordList = deepCopyList(recordList, x -> x.deepCopy());
+        AttendanceRecord toAdd = updatedRecordList.get(weekIndex).addAttendance(studentUuid, attendance);
+        updatedRecordList.set(weekIndex, toAdd);
+        return new AttendanceRecordList(updatedRecordList);
+    }
+
+    /**
+     * Edits a {@code Student}'s {@code Attendance} in the specified {@code Week}.
+     */
+    public AttendanceRecordList editAttendance(Student student, Week week, Attendance attendance)
+            throws InvalidWeekException, AttendanceNotFoundException {
+        requireAllNonNull(student, week, attendance);
+
+        if (!isWeekContained(week)) {
+            throw new InvalidWeekException();
+        }
+
+        UUID studentUuid = student.getUuid();
+        int weekIndex = week.getZeroBasedWeekNumber();
+        List<AttendanceRecord> updatedRecordList = deepCopyList(recordList, x -> x.deepCopy());
+        AttendanceRecord toEdit = updatedRecordList.get(weekIndex).editAttendance(studentUuid, attendance);
+        updatedRecordList.set(weekIndex, toEdit);
+        return new AttendanceRecordList(updatedRecordList);
+    }
+
+    /**
+     * Deletes a {@code Student}'s {@code Attendance} in the specified {@code Week}.
+     */
+    public AttendanceRecordList deleteAttendance(Student student, Week week)
+            throws InvalidWeekException, AttendanceNotFoundException {
+        requireAllNonNull(student, week);
+
+        if (!isWeekContained(week)) {
+            throw new InvalidWeekException();
+        }
+
+        UUID studentUuid = student.getUuid();
+        int weekIndex = week.getZeroBasedWeekNumber();
+        List<AttendanceRecord> updatedRecordList = deepCopyList(recordList, x -> x.deepCopy());
+        AttendanceRecord toDelete = updatedRecordList.get(weekIndex).deleteAttendance(studentUuid);
+        updatedRecordList.set(weekIndex, toDelete);
+        return new AttendanceRecordList(updatedRecordList);
+    }
```

Comment:
Similarly, these methods should be reserved for the commands in logic to handle to better decouple the model from the controller/logic.